webapp/src/app.tsx

```diff
@@ -5613,6 +5613,11 @@ document.addEventListener("DOMContentLoaded", async () => {
         await auth.loginCallbackAsync(query);
     }
 
+    // Handle logins that start via query parameter (for now, this is only Clever)
+    if (query["authlogin"]) {
+        await auth.loginAsync(query["authlogin"] as pxt.IdentityProviderId, true);
```

Comment:
This could cause a potential runtime issue.

I tried to build out our `IdentityProviderId` type in a smarter way so I could verify its value was a valid IDP. See the commit labeled [optional] for details on what I did. I described my issue with that [here](https://github.com/microsoft/pxt/pull/9458#discussion_r1163303146).

Reply:
Turns out it wouldn't throw, and the value would be passed in as-is regardless. `as pxt.IdentityProviderId` is just to make typescript happy.